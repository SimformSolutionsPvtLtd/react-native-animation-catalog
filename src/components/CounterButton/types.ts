export interface CounterButtonProps {
  label: string;
}
