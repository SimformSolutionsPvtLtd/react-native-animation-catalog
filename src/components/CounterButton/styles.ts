import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default styles;
