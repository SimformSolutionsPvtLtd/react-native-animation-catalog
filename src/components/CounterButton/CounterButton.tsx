import * as React from 'react';
import { Button, Text, View } from 'react-native';
import styles from './styles';
import type { CounterButtonProps } from './types';

export const updateCount = (input: number) => input + 1;

export const CounterButton = ({ label = 'Press Me' }: CounterButtonProps) => {
  const [count, setCount] = React.useState(0);
  return (
    <View style={styles.container}>
      <Text>You pressed {count} times</Text>
      <Button onPress={() => setCount(updateCount(count))} title={label} />
    </View>
  );
};

export default CounterButton;
