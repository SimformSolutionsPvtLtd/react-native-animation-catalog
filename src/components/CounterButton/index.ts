import CounterButton from './CounterButton';

export { CounterButton };
export type { CounterButtonProps } from './types';
