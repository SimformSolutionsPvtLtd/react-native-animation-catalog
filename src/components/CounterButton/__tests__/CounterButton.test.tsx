//@ts-nocheck
import React from 'react';
import { render } from '@testing-library/react-native';
import { CounterButton } from '../CounterButton';

jest.useFakeTimers();

describe('CounterButton component', () => {
  it('Match Snapshot', () => {
    const { toJSON } = render(<CounterButton />);
    expect(toJSON()).toMatchSnapshot();
  });
});
