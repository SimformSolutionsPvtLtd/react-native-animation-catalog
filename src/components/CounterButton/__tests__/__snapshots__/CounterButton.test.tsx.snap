// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component Match Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    You pressed 
    0
     times
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
